File: 2022/ruby/lib/day01.rb
# frozen_string_literal: true

class Day01
  def initialize(input)
    @input = input
  end

  def calories
    @input
      .split("\n\n")
      .map { |str| str.lines.map(&:to_i) }
  end

  def most_calories
    calories_agg_desc.first
  end

  def top3_calories
    calories_agg_desc.take(3).sum
  end

  private

    def calories_agg_desc
      calories.map(&:sum).sort.reverse
    end
end

File: 2022/clojure/day01.clj
(require '[clojure.string :as str])

(def input "1
2

3
4

5")

(def consume (fn [str] (->>
                        (str/split str #"\n\n")
                        (map #(->>
                               (str/split %1 #"\n")
                               (map read-string)
                               (reduce +)))
                        (sort)
                        (reverse))))

(defn consume2 [str] (->>
                      (str/split str #"\n\n")
                      (map #(->>
                             (str/split %1 #"\n")
                             (map read-string)
                             (reduce +)))
                      (sort)
                      (reverse)))

(consume2 input)
(ns day01
  (:require [clojure.test :refer [deftest, is]]))

(deftest addition-tests
  (is (= 5 (+ 3 2)))
  (is (= 10 (+ 5 5))))

